### PR TITLE
Cleanup built binary downloading

### PR DESCRIPTION
**Pull request by hennevogel**
There were several layers of stuff on top of a simple url. If a package is published, the backend knows best where it ended up. If we don't publish we can simply point to the API url for the file.

No need for a Service that checks existence of files over the network, rewrites hostnames or an extra route that prefixes the API path with `/package`.

### CODE REVIEW

**Comment by hennevogel** [comment]
Depends on #13103

File: src/api/app/controllers/webui/package_controller.rb [hunk 1]
```diff
@@ -4,22 +4,22 @@ class Webui::PackageController < Webui::WebuiController
   include Webui::ManageRelationships
   include BuildLogSupport
 
+  NON_BINARY_BUILD_RESTULS = ['rpmlint.log', '_buildenv'].freeze
```
Comment (danidoni) [inline src/api/app/controllers/webui/package_controller.rb:7]:
```suggestion
  NON_BINARY_BUILD_RESULTS = ['rpmlint.log', '_buildenv'].freeze
```

File: src/api/app/controllers/webui/package_controller.rb [hunk 2]
```diff
@@ -182,24 +182,22 @@ def statistics
...
-    logger.debug "accepting #{request.accepts.join(',')} format:#{request.format}"
-    # little trick to give users eager to download binaries a single click
-    redirect_to(@download_url) && return if request.format != Mime[:html] && @download_url
+    @download_url = @package.download_url_for_binary(repository_name: @repository.name,
+                                                     architecture_name: @architecture.name,
+                                                     package_name: @package_name,
+                                                     filename: @filename)
   end
 
+  # FIXME: This is Webui::Packages::BinariesController#index
```
Comment (danidoni) [inline src/api/app/controllers/webui/package_controller.rb:200]: I don't understand this comment, can you explain a bit?

File: src/api/app/controllers/webui/package_controller.rb [hunk 3]
```diff
@@ -208,16 +206,22 @@ def binaries
...
         else
-          links = links_for_binaries_action(@project, @package_name, repository, result['arch'], binary['filename'])
-          build_results_set[:binaries] << { filename: binary['filename'], size: binary['size'], links: links }
+          download_url = @package.download_url_for_binary(repository_name: @repository,
+                                                          architecture_name: result['arch'],
+                                                          package_name: @package_name, # multibuild flavors...
+                                                          filename: binary['filename'])
+          build_results_set[:binaries] << { filename: binary['filename'],
+                                            size: binary['size'],
+                                            links: { details?: NON_BINARY_BUILD_RESTULS.exclude?(binary['filename']),
```
Comment (danidoni) [inline src/api/app/controllers/webui/package_controller.rb:222]:
```suggestion
                                            links: { details?: NON_BINARY_BUILD_RESULTS.exclude?(binary['filename']),
```

File: src/api/app/controllers/webui/package_controller.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb:200]: This should be extracted into it's own controller...

File: src/api/app/controllers/webui/package_controller.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb:7]: This reminds me, I wanted to turn this around...

Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb]: done

File: src/api/config/options.yml.example [hunk 4]
```diff
@@ -26,11 +26,6 @@ default: &default
   frontend_host: localhost
   frontend_port: 443
   frontend_protocol: https
-  # if your users access the hosts through a proxy (or just a different name, use this to
-  # overwrite the settings for users)
-  #external_frontend_host: api.opensuse.org
-  #external_frontend_port: 443
-  #external_frontend_protocol: https
```
Comment (eduardoj) [inline src/api/config/options.yml.example:33]: There are some other lines that could be removed from `dist/functions.setup-appliance.sh`:
```bash
# grep -r external_frontend_ dist/
dist/functions.setup-appliance.sh:      sed -i 's,^external_frontend_host: .*,frontend_host: "'"$FQHOSTNAME"'",' $api_options_yml
dist/functions.setup-appliance.sh:      sed -i 's,^external_frontend_port: .*,frontend_port: 443,' $api_options_yml
dist/functions.setup-appliance.sh:      sed -i 's,^external_frontend_protocol: .*,frontend_protocol: "'"https"'",' $api_options_yml
```

Comment (hennevogel) [reply src/api/config/options.yml.example:33]: :+1: done

**Review by danidoni — APPROVED** [review]

**Pull request merged**